Remotion/Web/UnitTests/Core/ExecutionEngine/UrlMapping/CompoundUrlMappingFileFinderTest.cs

```diff
@@ -0,0 +1,89 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System.Collections.Generic;
+using Moq;
+using NUnit.Framework;
+using Remotion.Web.ExecutionEngine.UrlMapping;
+
+namespace Remotion.Web.UnitTests.Core.ExecutionEngine.UrlMapping
+{
+  [TestFixture]
+  public class CompoundUrlMappingFileFinderTest
+  {
+    private class UrlMappingFileFinderStub : IUrlMappingFileFinder
+    {
+      private readonly string[] _values;
+
+      public UrlMappingFileFinderStub (params string[] values)
+      {
+        _values = values;
+      }
+
+      public IEnumerable<string> GetUrlMappingFilePaths ()
+      {
+        return _values;
+      }
+    }
+
+    [Test]
+    public void Initialize ()
+    {
+      var urlMappingFileFinders = new[] { new Mock<IUrlMappingFileFinder>().Object, new Mock<IUrlMappingFileFinder>().Object };
+
+      var compoundFactory = new CompoundUrlMappingFileFinder(urlMappingFileFinders);
+
+      Assert.That(compoundFactory.UrlMappingFileFinders, Is.Not.SameAs(urlMappingFileFinders));
+      Assert.That(compoundFactory.UrlMappingFileFinders, Is.EqualTo(urlMappingFileFinders));
+    }
+
+    [Test]
+    public void GetQueryFilePaths ()
+    {
+      var urlMappingFileFinders = new IUrlMappingFileFinder[]
+                             {
+                                 new UrlMappingFileFinderStub(@"C:\query1.xml", @"C:\query2.xml"),
+                                 new UrlMappingFileFinderStub(@"C:\Query3.xml"),
+                             };
+      var compoundQueryFileFinder = new CompoundUrlMappingFileFinder(urlMappingFileFinders);
+
+      var expectedQueryFiles = new[]
+                               {
+                                   @"C:\query1.xml",
+                                   @"C:\query2.xml",
+                                   @"C:\Query3.xml"
+                               };
+      Assert.That(compoundQueryFileFinder.GetUrlMappingFilePaths(), Is.EqualTo(expectedQueryFiles));
+    }
+
+    [Test]
+    public void GetQueryFilePaths_RemovesDuplicateQueryFilePaths ()
```

Comment:
gut feeling for tests: in distinct tests I try to always also have a second value, or perhaps two distinct groups that result in two values at the end to ensure / make obvious there's no chance of an "take first" error in the code. Basically, write the test set in such a way that it's impossible to make this false positive without maliciously incorrect ode while still keeping it short.